test/request_test.go

```diff
@@ -1,40 +1,154 @@
 package main
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
+	"os"
 	"testing"
-  "io/ioutil"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/vkuznet/transfer2go/core"
+	"github.com/vkuznet/transfer2go/utils"
 )
 
 var url = "http://localhost:8989"
 
-func TestGetReposHandler(t *testing.T) {
+type tests struct {
+	description        string
+	url                string
+	expectedStatusCode int
+	expectedBody       string
+	result             bool
+}
+
+// Function names are according to api endpoints
+
+func TestStatus(t *testing.T) {
+	assert := assert.New(t)
+
+	test := tests{
+		description:        "Check status of server",
+		url:                url + "/status",
+		expectedStatusCode: 200,
+		expectedBody:       "Test",
+	}
+
+	var data map[string]interface{}
+
+	resp, err := http.Get(test.url)
+	actual, err := ioutil.ReadAll(resp.Body)
+
+	defer resp.Body.Close()
+	json.Unmarshal([]byte(actual), &data)
+	assert.NoError(err)
+
+	assert.Equal(test.expectedStatusCode, resp.StatusCode, test.description)
+	assert.Equal(test.expectedBody, data["name"], test.description)
+
+}
+
+func TestAgents(t *testing.T) {
+	assert := assert.New(t)
+
+	test := tests{
+		description:        "Test the list of connected agents",
+		url:                url + "/agents",
+		expectedStatusCode: 200,
+		expectedBody:       "http://localhost:8989",
+	}
+
+	var data map[string]interface{}
+
+	resp, err := http.Get(test.url)
+	actual, err := ioutil.ReadAll(resp.Body)
+
+	defer resp.Body.Close()
+	json.Unmarshal([]byte(actual), &data)
+	assert.NoError(err)
+
+	assert.Equal(test.expectedStatusCode, resp.StatusCode, test.description)
+	assert.Equal(test.expectedBody, data["Test"], test.description)
+
+}
+
+func TestWriteTFC(t *testing.T) {
+	assert := assert.New(t)
+
+	test := tests{
+		description:        "Check TFC upload functionality",
+		url:                url + "/tfc",
+		expectedStatusCode: 200,
+		expectedBody:       "http://localhost:8989",
+	}
+
+	f, err := os.Create("data/testdata.txt")
```

Comment:
I created separate space to store the test data configuration. I think so it will be good if we create this testdata.txt inside that folder.